File: packages/swift/Sources/CatalystSwift/Models/Pagination.swift
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

/// Total/offset counters used by several list endpoints.
public struct CatalystCountInfo: Decodable, Sendable, Equatable, Hashable {
  public let total: Int
  public let offset: Int
}

/// Page cursor metadata used by several list endpoints.
public struct CatalystPageInfo: Decodable, Sendable, Equatable, Hashable {
  public let min: Int
  public let max: Int
  public let current: Int
  public let next: Int?
  public let prev: Int?
}

File: packages/kotlin/src/main/kotlin/com/natsuneko/catalyst/models/Pagination.kt
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

package com.natsuneko.catalyst.models

import kotlinx.serialization.Serializable

/**
 * Total/offset counters used by several list endpoints
 */
@Serializable
data class CatalystCountInfo(
    val total: Int,
    val offset: Int
)

/**
 * Page cursor metadata used by several list endpoints
 */
@Serializable
data class CatalystPageInfo(
    val min: Int,
    val max: Int,
    val current: Int,
    val next: Int? = null,
    val prev: Int? = null
)
